KarhooSDK/Network/HttpClient/TokenRefreshingHttpClient/Interactor/KarhooTokenRefreshNeedWorker.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  KarhooTokenRefreshNeedWorker.swift
+//  KarhooSDK
+//
+//  Created by Aleksander Wedrychowski on 05/10/2022.
+//  Copyright © 2022 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+
+protocol TokenRefreshNeedWorker: AnyObject {
+    func saveRefreshBuffer(token: AuthToken)
+    func tokenNeedsRefreshing(credentials: Credentials) -> Bool
+    func refreshTokenNeedsRefreshing(credentials: Credentials?) -> Bool
+}
+
+final class KarhooTokenRefreshNeedWorker: TokenRefreshNeedWorker {
+    private enum Constants {
+        /// Seconds buffer when refresh token should be refreshed proactively, so it never becomes expired. Default value 5 mins. Updated each time new AuthToken is saved.
+        static var refreshBuffer: TimeInterval = 5 * 60
```

Comment:
Didn't we agree to lose the 5 mins completely? @mateid-kr correct me if I'm wrong please

Reply:
This is default value until we get any Token fetched